python/SubstraitDialects.cpp

```diff
@@ -7,19 +7,23 @@
 //===----------------------------------------------------------------------===//
 
 #include "mlir-c/BuiltinAttributes.h"
-#include "mlir-c/BuiltinTypes.h"
 #include "mlir-c/IR.h"
-#include "mlir/Bindings/Python/PybindAdaptors.h"
+#include "mlir/Bindings/Python/NanobindAdaptors.h" // IWYU pragma: keep
 #include "substrait-mlir-c/Dialects.h"
 #include "llvm/Support/FileSystem.h"
 #include "llvm/Support/Signals.h"
 
-#include <vector>
+// Suppress warning from nanobind, otherwise CI with `-Werror` fails.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wcast-qual"
+#pragma clang diagnostic ignored "-Wnested-anon-types"
+#pragma clang diagnostic ignored "-Wzero-length-array"
+#include <nanobind/nanobind.h>
```

Comment:
If you include mlir/include/mlir/Bindings/Python/Nanobind.h instead, I've already done similar squashing so that you don't have to think here.

Reply:
Yeah, nice, that's better! Thanks for the tip :)